### PR TITLE
[repacker] Extension promotion mechanism in the repacker

### PR DESCRIPTION
**Pull request by garretrieger**
When enabled analyzes the non-extension lookups within a GSUB/GPOS table and decides if any need to be promoted into extensions prior to space assignemnt.

This is v1 of the algorithm which doesn't factor in shared nodes between lookups.

### CODE REVIEW

File: src/hb-ot-layout-gsubgpos.hh [hunk 1]
```diff
@@ -3988,7 +3988,7 @@ struct GSUBGPOSVersion1_2
 {
   friend struct GSUBGPOS;
 
-  protected:
+  public: // TODO
```
Comment (behdad) [inline src/hb-ot-layout-gsubgpos.hh]: Add a friend statement if you can. Otherwise fine with me.

**Comment by behdad** [comment]
I'm not very well-versed in this code. Looks good to me in general.

File: src/hb-ot-layout-gsubgpos.hh (hunk 1 above)
Comment (garretrieger) [reply src/hb-ot-layout-gsubgpos.hh]: Ended up just adding a method to access the field I needed.

**Comment by behdad** [comment]
Is this ready now?

**Comment by garretrieger** [comment]
Yeah

**Pull request merged**